File: sandbox/js/create_mock_endpoint_proxy.mjs
import { EndpointProxy } from "../../packages/js/endpoint_proxy.mjs";
import { ResultHandoff } from "../../packages/js/result_handoff.mjs";
import { StateSync } from "../../packages/js/state_sync.mjs";
import { MockEnvelopeEndpoint } from "./envelope_endpoint_mock.mjs";
import { MockJPItemQueue } from "./jpitem_queue_mock.mjs";
import { TransportLayerMock } from "./transport_layer_mock.mjs";

export function createMockEndpointProxy() {
  return new EndpointProxy({
    stateSync: new StateSync({
      transportLayer: new TransportLayerMock(),
    }),
    resultHandoff: new ResultHandoff({
      transportLayer: new TransportLayerMock(),
    }),
    queue: new MockJPItemQueue(new MockEnvelopeEndpoint()),
  });
}

File: sandbox/python/jobping_sandbox/create_mock_endpoint_proxy.py
from jobping.endpoint_proxy import EndpointProxy
from jobping.result_handoff import ResultHandoff
from jobping.state_sync import StateSync
from jobping.imp.envelope_endpoint_inmemory import EnvelopeEndpointInMemory
from jobping.imp.jpitem_queue_inmemory import JPItemQueueInMemory
from jobping_sandbox.transport_layer_mock import TransportLayerMock


def create_mock_endpoint_proxy() -> EndpointProxy:
    return EndpointProxy(
        state_sync=StateSync(TransportLayerMock()),
        result_handoff=ResultHandoff(TransportLayerMock()),
        queue=JPItemQueueInMemory(EnvelopeEndpointInMemory()),
    )
